### DISCUSSION TITLE
POST Request for Docker Compose File in GitHub Actions

### DISCUSSION THREAD

**Discussion by allgandalf** [community/community#57645] [Actions] [answered]
### Select Topic Area

Bug

### Body

I am running services from a docker compose file and trying to test it using github actions. I am able to make a [GET request](https://github.com/RohanSasne/pecan/actions/runs/5223587597/jobs/9430653728#step:9:1) successfully, but I am not able to make a [POST request](https://github.com/RohanSasne/pecan/actions/runs/5223446882/jobs/9430317172#step:9:7), it redirects me to a wrong url.Also it first gives me a 307 status code before being redirected.

The post request is running totally fine in my local system. The response screenshot is as follows: ![enter image description here](https://i.stack.imgur.com/xOqKa.png)

**Comment by airtower-luna** [comment]
Here's the difference: https://github.com/RohanSasne/pecan/actions/runs/5223446882/jobs/9430317172#step:9:55

In your screenshot, the `Location` header of the redirect gives a relative URL, so curl queries the same host as before. In your Actions run it points at a full URL with the hostname `pecan.localhost`, which naturally points at the local host (the connection to localhost then fails) instead of the previous IP (I assume a container). You'll have to check your code to figure out why it issues the different redirect.

**Reply by allgandalf** [reply]
I am getting confused as to why the change occurs when running it on GitHub actions, cause locally things are great, but in GitHub actions, there seem to be some configuration mismatch i suppose, thanks for pointing out though, if you are able to find a solution to this one, it would be great :)

**Comment by allgandalf** [comment, marked as answer]
Well it turns out one of the service was crashing after initial startup, solved the issue now.

For people facing the same issue in the future, set the service to restart until it's manually stopped: `restart: unless-stopped`
